src/compute/src/render/context.rs

```diff
@@ -97,6 +98,24 @@ where
     /// The expiration time for dataflows in this context. The output's frontier should never advance
     /// past this frontier, except the empty frontier.
     pub dataflow_expiration: Antichain<T>,
+    /// The flags active in this context.
+    pub flags: ContextFlags,
```

Comment:
Have you considered just passing a `ConfigSet` directly? That would reduce some boilerplate when adding a new config. You could simply access it everywhere that has access to a `Context`, without needing to change the fields and `From` implement of `ContextFlags`. Also makes it possible to dynamically change some configuration at runtime (without recreating the dataflow), should we ever want that.

Reply:
Great point! Changed it such that we just have a single `Rc<ConfigSet>` (updating doesn't require mutability), which seems better than introducing a custom struct.